selfservice/flow/login/hook.go

```diff
@@ -133,7 +133,11 @@ func (e *HookExecutor) PostLoginHook(
 	r = r.WithContext(ctx)
 	defer otelx.End(span, &err)
 
-	if err := e.maybeLinkCredentials(r.Context(), s, i, f); err != nil {
+	// We need to set the identity here because we check the available AAL in maybeLinkCredentials.
+	s.IdentityID = i.ID
+	s.Identity = i
+
+	if err := e.maybeLinkCredentials(ctx, s, i, f); err != nil {
```

Comment:
For me, the `PostLoginHook` in general feels a bit weird with all the order of operations.

1. webhooks fire after first factor even when 2fa is required: maybe useful?
2. 2fa requirement is not checked in the beginning or after the hook fired, but somewhere nested. I would suggest to have that as one of the very first checks.